From 5a13a97c2112ec58d5171b8cbbe4c85e8570c56d Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Thu, 21 Dec 2023 12:54:37 +0100
Subject: [PATCH] com.webos.service.mediacontroller: Make it work generically

Not liking the hardcoded bits from OSE, let's make it more flexible and keep the same functionality
---
Upstream-Status: Pending

 CMakeLists.txt              |  6 ++----
 src/MediaControlService.cpp | 33 +++++++++++++--------------------
 src/MediaSessionManager.cpp |  2 +-
 3 files changed, 16 insertions(+), 25 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 13f158f..6434b6e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,12 +45,10 @@ if(CMAKE_BUILD_TYPE STREQUAL "Debug")
 add_definitions(-DUSE_TEST_METHOD)
 endif()
 
-if(WEBOS_TARGET_MACHINE STREQUAL "raspberrypi4" OR WEBOS_TARGET_MACHINE STREQUAL "raspberrypi4-64" OR WEBOS_TARGET_MACHINE STREQUAL "qemux86" OR WEBOS_TARGET_MACHINE STREQUAL "qemux86-64")
-add_definitions(-DPLATFORM_RASPBERRYPI4)
-endif()
-
 if(WEBOS_TARGET_MACHINE STREQUAL "sa8155")
 add_definitions(-DPLATFORM_SA8155)
+else ()
+add_definitions(-DPLATFORM_GENERIC)
 endif()
 
 webos_add_compiler_flags(ALL -Wall -funwind-tables)
diff --git a/src/MediaControlService.cpp b/src/MediaControlService.cpp
index 5faecac..b9a140c 100644
--- a/src/MediaControlService.cpp
+++ b/src/MediaControlService.cpp
@@ -150,14 +150,7 @@ bool MediaControlService::onBTAdapterQueryCb(LSHandle *lshandle, LSMessage *mess
       std::string adapterAddress = adapters[i]["adapterAddress"].asString();
       PMLOG_INFO(CONST_MODULE_MCS,"%s : adapterName : %s, adapterAddress : %s",
                                    __FUNCTION__, adapterName.c_str(), adapterAddress.c_str());
-
-#if defined(PLATFORM_RASPBERRYPI4)
-      if(("raspberrypi4 #2" == adapterName) || ("raspberrypi4" == adapterName) ||
-         ("raspberrypi4-64 #2" == adapterName) || ("raspberrypi4-64" == adapterName) ||
-         ("qemux86-64 #2" == adapterName) || ("qemux86-64" == adapterName)) {
-#elif defined(PLATFORM_SA8155)
-      if ("sa8155 Bluetooth hci0" == adapterName) {
-#endif
+      if (("" != adapterName) && ("sa8155 Bluetooth hci1" != adapterName)) {
         deviceSetId = "RSE-L";
         std::string payload = "{\"adapterAddress\":\"" + adapterAddress + "\",\"subscribe\":true}";
         PMLOG_INFO(CONST_MODULE_MCS,"%s : payload = %s for first adapter",__FUNCTION__, payload.c_str());
@@ -183,9 +176,9 @@ bool MediaControlService::onBTAdapterQueryCb(LSHandle *lshandle, LSMessage *mess
             PMLOG_ERROR(CONST_MODULE_MCS,"%s : LSCall failed for BT device/getStatus", __FUNCTION__);
         }
       }
-      else
-        PMLOG_ERROR(CONST_MODULE_MCS,"%s: Already subscribe for adapter: %s", __FUNCTION__, adapterAddress.c_str());
-
+      else {
+        PMLOG_ERROR(CONST_MODULE_MCS,"%s: Already subscribed for adapter: %s", __FUNCTION__, adapterAddress.c_str());
+      }
     PMLOG_INFO(CONST_MODULE_MCS,"%s : adapterAddress = %s deviceSetId = %s [%d]",__FUNCTION__, adapterAddress.c_str(), deviceSetId.c_str(), __LINE__);
     if (obj && !deviceSetId.empty() && !adapterAddress.empty())
       obj->ptrMediaControlPrivate_->setBTAdapterInfo(deviceSetId, adapterAddress);
@@ -281,7 +274,7 @@ bool MediaControlService::onBTAvrcpGetStatusCb(LSHandle *lshandle, LSMessage *me
 
         if(!(obj->ptrMediaControlPrivate_->isDeviceRegistered(address, adapterAddress))) {
           int displayId = obj->ptrMediaControlPrivate_->getDisplayIdForBT(adapterAddress);
-#if defined(PLATFORM_RASPBERRYPI4)
+#if defined(PLATFORM_GENERIC)
           displayId = 0;
 #endif
           BTDeviceInfo objDevInfo(address, adapterAddress, "", displayId);
@@ -371,7 +364,7 @@ bool MediaControlService::onBTAvrcpKeyEventsCb(LSHandle *lshandle, LSMessage *me
         std::string mediaId = obj->ptrMediaControlPrivate_->getMediaId(address);
         int displayIdForMedia = obj->ptrMediaSessionMgr_->getDisplayIdForMedia(mediaId);
         //ToDo : Below platform check to be removed once multi intsnace support available in chromium for OSE
-#if defined(PLATFORM_RASPBERRYPI4)
+#if defined(PLATFORM_GENERIC)
         displayIdForBT = displayIdForMedia = 0;
 #endif
         PMLOG_INFO(CONST_MODULE_MCS, "%s mediaId for sending BT key event : %s", __FUNCTION__, mediaId.c_str());
@@ -866,7 +859,7 @@ bool MediaControlService::setMediaMetaData(LSMessage& message) {
       /*Get display ID from media ID*/
       int displayId = ptrMediaSessionMgr_->getDisplayIdForMedia(mediaId);
       //ToDo : Below platform check to be removed once dual blueetooth support in OSE
-#if defined(PLATFORM_RASPBERRYPI4)
+#if defined(PLATFORM_GENERIC)
       displayId = 0;
 #endif
       pbnjson::JObject metaDataObj;
@@ -951,7 +944,7 @@ bool MediaControlService::setMediaPlayStatus(LSMessage& message) {
         PMLOG_INFO(CONST_MODULE_MCS, "%s Invalid PlaybackStatus", __FUNCTION__);
 
       int displayIdForMedia = ptrMediaSessionMgr_->getDisplayIdForMedia(mediaId);
-#if defined(PLATFORM_RASPBERRYPI4)
+#if defined(PLATFORM_GENERIC)
       displayIdForMedia = 0;
 #endif
       BTDeviceInfo objDevInfo;
@@ -986,7 +979,7 @@ bool MediaControlService::setMediaPlayStatus(LSMessage& message) {
       /*Get display ID from media ID*/
       int displayId = ptrMediaSessionMgr_->getDisplayIdForMedia(mediaId);
       //ToDo : Below platform check to be removed once dual blueetooth support in OSE
-#if defined(PLATFORM_RASPBERRYPI4)
+#if defined(PLATFORM_GENERIC)
       displayId = 0;
 #endif
       pbnjson::JValue responsePayload = pbnjson::Object();
@@ -1056,7 +1049,7 @@ bool MediaControlService::setMediaMuteStatus (LSMessage & message){
       /*Get display ID from media ID*/
       int displayId = ptrMediaSessionMgr_->getDisplayIdForMedia(mediaId);
       //ToDo : Below platform check to be removed once dual blueetooth support in OSE
-#if defined(PLATFORM_RASPBERRYPI4)
+#if defined(PLATFORM_GENERIC)
       displayId = 0;
 #endif
       pbnjson::JValue responsePayload = pbnjson::Object();
@@ -1121,7 +1114,7 @@ bool MediaControlService::setMediaPlayPosition (LSMessage & message){
       /*Get display ID from media ID*/
       int displayId = ptrMediaSessionMgr_->getDisplayIdForMedia(mediaId);
       //ToDo : Below platform check to be removed once dual blueetooth support in OSE
-#if defined(PLATFORM_RASPBERRYPI4)
+#if defined(PLATFORM_GENERIC)
       displayId = 0;
 #endif
       pbnjson::JValue responsePayload = pbnjson::Object();
@@ -1174,7 +1167,7 @@ bool MediaControlService::receiveMediaPlaybackInfo (LSMessage & message){
   pbnjson::JValue payload = msg.get();
   int displayId  = payload["displayId"].asNumber<int>();
       //ToDo : Below platform check to be removed once dual blueetooth support in OSE
-#if defined(PLATFORM_RASPBERRYPI4)
+#if defined(PLATFORM_GENERIC)
       displayId = 0;
 #endif
   bool subscribed = payload["subscribe"].asBool();
@@ -1284,7 +1277,7 @@ bool MediaControlService::injectMediaKeyEvent(LSMessage &message) {
   pbnjson::JValue payload = msg.get();
   int displayId  = payload["displayId"].asNumber<int>();
       //ToDo : Below platform check to be removed once dual blueetooth support in OSE
-#if defined(PLATFORM_RASPBERRYPI4)
+#if defined(PLATFORM_GENERIC)
       displayId = 0;
 #endif
   std::string keyEvent = payload["keyEvent"].asString();
diff --git a/src/MediaSessionManager.cpp b/src/MediaSessionManager.cpp
index e5d7f47..96aedc7 100644
--- a/src/MediaSessionManager.cpp
+++ b/src/MediaSessionManager.cpp
@@ -261,7 +261,7 @@ std::string MediaSessionManager::getMediaIdFromDisplayId(const int& displayId) {
   for (const auto& itr : mapMediaSessionInfo_) {
     std::string appId = itr.second.getAppId();
     int appDisplayId = (appId.back()-48);
-#if defined(PLATFORM_RASPBERRYPI4)
+#if defined(PLATFORM_GENERIC)
     appDisplayId = 0;
 #endif
     if(appDisplayId == displayId){
